cmd/frontend/graphqlbackend/site_config_change_connection.go

```diff
@@ -0,0 +1,133 @@
+package graphqlbackend
+
+import (
+	"context"
+
+	"github.com/graph-gophers/graphql-go"
+	"github.com/graph-gophers/graphql-go/relay"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type SiteConfigurationChangeConnectionStore struct {
+	db database.DB
+}
+
+func (s *SiteConfigurationChangeConnectionStore) ComputeTotal(ctx context.Context) (*int32, error) {
+	count, err := s.db.Conf().GetSiteConfigCount(ctx)
+	c := int32(count)
+	return &c, err
+}
+
+func copyIntPtr(n *int) *int {
+	if n == nil {
+		return nil
+	}
+
+	c := *n
+	return &c
+}
+
+// modifyArgs will fetch one more than the originally requested number of items because we need one
+// older item to get the diff of the oldes item in the list.
+//
+// A separate function so that this can be tested in isolation.
+func modifyArgs(args *database.PaginationArgs) bool {
+	var modified bool
+	if args.First != nil {
+		*args.First += 1
+		modified = true
+	} else if args.Last != nil && args.Before != nil {
+		if *args.Before > 0 {
+			modified = true
+			*args.Last += 1
+			*args.Before -= 1
+		}
+	}
+
+	return modified
+}
+
+func (s *SiteConfigurationChangeConnectionStore) ComputeNodes(ctx context.Context, args *database.PaginationArgs) ([]*SiteConfigurationChangeResolver, error) {
+	if args == nil {
+		return []*SiteConfigurationChangeResolver{}, errors.New("pagintation args cannot be nil")
```

Comment:
I guess we may as well return nil here. + typo fix
```suggestion
		return nil, errors.New("pagination args cannot be nil")
```